Classes/Player/Media/SourceSelector.swift

```diff
@@ -51,12 +51,21 @@ class SourceSelector {
             }
         }
         
-        if DRMSupport.fairplay {
-            if let source = sources.first(where: {$0.mediaFormat == .hls}) {
-                return (source, defaultHandler.init())
-            }
-        } else {
-            if let source = sources.first(where: {$0.mediaFormat == .hls && ($0.drmData == nil || $0.drmData!.isEmpty) }) {
+        if let source = sources.first(where: {$0.mediaFormat == .hls}) {
```

Comment:
I think we need to have clear handling in player.
If app wants to play DRM it needs to provide sources with only DRM and if player fails cause it can't support it then it should fail.
If the app wants to fallback to clear it can listen to the error event and re re-prepare the player with clear sources.

Reply:
Usually they set the _formats_ or/and _fileIds_ in order to fetch the required media/type, so usually there aren't a mixture of types.
It was always like this and the following is a comment in the code:
"// Preference: Local, HLS, FPS*, MP4, WVM*, MP3"

We will need to discuss it for a change and how to do it without breaking it.

Reply:
This case covered in another PR #425